File: confinder.application/Models/ConferenceListRequest.cs
using System;

namespace confinder.application.Models
{
    public class ConferenceListRequest
    {
        public SortField? SortField { get; set; }
        public SortOrder? SortOrder { get; set; }
        public int? Page { get; set; }

		public string? Name { get; set; }
		public string? Location { get; set; }
        public string? MinQualisIndex { get; set; }
        public string? MaxQualisIndex { get; set; }
        public DateOnly? MinEventDate { get; set; }
        public DateOnly? MaxEventDate { get; set; }
        public DateOnly? MinSubmissionDeadline { get; set; }
        public DateOnly? MaxSubmissionDeadline { get; set; }
    }
}

File: confinder.web/src/types/ConferenceListRequest.ts
import { QualisIndex } from './QualisIndex';

export type ConferenceListRequest = {
  page: number;
  name?: string;
  location?: string;
  minQualisIndex?: QualisIndex;
  maxQualisIndex?: QualisIndex;
  minEventDate?: Date;
  maxEventDate?: Date;
  minSubmissionDeadline?: Date;
  maxSubmissionDeadline?: Date;
};
